packages/side-nav/src/SideNavGroupCollapsed/SideNavGroupCollapsed.tsx

```diff
@@ -0,0 +1,122 @@
+
+import { useSideNavContext } from '../SideNav/SideNavContext';
+import { ulStyleOverrides } from '../SideNav/styles';
+import {
+  baseStyle,
+  buttonClassName,
+  indentedStyle,
+  themeStyle,
+} from '../SideNavGroup/styles';
+
+import {
+  collapsibleBaseStyle,
+  collapsibleFocusStyle,
+  collapsibleGroupBaseStyles,
+  collapsibleThemeStyle,
+  expandIconStyle,
+  openExpandIconStyle,
+  transitionStyles,
+  ulStyles,
+  ulTransitionStyles,
+} from './styles';
+import { SideNavGroupCollapsedProps } from './types';
+
+/**
+ * @internal
+ */
+export function SideNavGroupCollapsed({
+  groupHeaderProps,
+  open,
+  setOpen,
+  menuGroupLabelId,
+  indentLevel,
+  renderHeader,
+  children,
+}: SideNavGroupCollapsedProps) {
+  const { width, theme, darkMode } = useSideNavContext();
+  const { usingKeyboard } = useUsingKeyboardContext();
+
+  const menuId = useIdAllocator({ prefix: 'menu' });
+
+  const nodeRef = React.useRef(null);
+  const ulRef = React.useRef<HTMLUListElement>(null);
+
+  // compute the entered ul wrapper styles based on the ul height
+  useEffect(() => {
+    const ulHeight = ulRef?.current?.getBoundingClientRect().height ?? 0;
+    transitionStyles['entered'] = css`
+      opacity: 1;
+      max-height: ${ulHeight + 1}px; // +1 for border
+      border-bottom: 1px solid
+        ${darkMode ? palette.gray.dark1 : palette.gray.light2};
+    `;
+  }, [open, ulRef, darkMode]);
+
+  return (
+    <>
+      <button
+        {...groupHeaderProps}
+        aria-controls={menuId}
+        aria-expanded={open}
+        className={cx(
+          buttonClassName,
+          baseStyle,
+          themeStyle[theme],
+          collapsibleBaseStyle,
+          collapsibleThemeStyle[theme],
+          css`
+            width: ${width}px;
+          `,
+          {
+            [collapsibleFocusStyle]: usingKeyboard,
+            [indentedStyle(indentLevel, darkMode)]: indentLevel > 1,
+          },
+        )}
+        onClick={() => setOpen((curr: boolean) => !curr)}
+      >
+        {renderHeader()}
+        <ChevronRight
+          role="presentation"
+          size={12}
+          className={cx(expandIconStyle, {
+            [openExpandIconStyle]: open,
+          })}
+        />
+      </button>
+      <Transition
+        in={open}
+        appear
+        timeout={150}
+        nodeRef={nodeRef}
+        mountOnEnter
+        unmountOnExit
+      >
+        {(state: TransitionStatus) => (
+          <div
+            ref={nodeRef}
+            className={cx(collapsibleGroupBaseStyles, transitionStyles[state])}
+          >
+            <ul
+              ref={ulRef}
+              id={menuId}
+              aria-labelledby={menuGroupLabelId}
+              className={cx(ulStyleOverrides, ulStyles, {
+                [ulTransitionStyles]: ['entering', 'entered'].includes(state),
+              })}
+            >
+              {children}
+            </ul>
+          </div>
+        )}
+      </Transition>
```

Comment:
Does this `Transition` belong in this subcomponent? I'm wondering because this means this component has to take in an `open` prop when the name is `SideNavCollapsed`, which feels a little weird. 

Removing this from here would also allow us to simply extend `HTMLElementProps<HTMLButtonElement>` here

Reply:
The name of this component could be better. `SideNavOpen` shows a header name and then underneath that is a menu with a number of children that are visible, `SideNavCollapsed` is the same except the menu is hidden and toggled open by clicking the header name. So yeah, the transition should be in this subcomponent because it controls the transition of the toggling. 


This component shows a header element and has a collapsible container that shows are the

Reply:
Maybe I change this to `SideNavGroupToggle`?